sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -731,7 +731,7 @@ def state(self):
       A PipelineState object.
     """
     if not self.has_job:
-      return PipelineState.UNKNOWN
```

Comment:
I don't think this is correct. self.has_job being false is a symptom suggesting support in fetching job info is incomplete.

Reply:
im targeting for this case: https://github.com/Triple-Whale/beam/blob/094eb7dc5d7bc9f3de9ef57e8d764f0d8defe037/sdks/python/apache_beam/runners/dataflow/dataflow_runner.py#L480

since this is in the DataflowPipelineResult i think its safe to assume this is only used on completion.

Reply:
Thanks, however I was not familiar with this code path. R: @robertwb for double check

Reply:
Is the issue here that we need a terminal (presumably non-failure) state? 

I think this is OK--we are done uploading the template, but have no idea what the state of the job(s) is/are. It's worth a comment at least (both here and where we set it indicating that a None job is interpreted as success).

Reply:
comment exists, here's where i took it from: https://github.com/apache/beam/blob/8f71dc41b30a978095ca0e0699009e4f4445a618/sdks/python/apache_beam/runners/dataflow/dataflow_runner.py#L867-L870